onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.cc

```diff
@@ -121,4 +127,307 @@ std::shared_ptr<KernelRegistry> MKLDNNExecutionProvider::GetKernelRegistry() con
   static std::shared_ptr<KernelRegistry> kernel_registry = onnxruntime::mkl_dnn::GetMklDnnKernelRegistry();
   return kernel_registry;
 }
+
+bool MKLDNNExecutionProvider::UseSubgraph(const onnxruntime::GraphViewer& graph_viewer,
+                                          const std::vector<const KernelRegistry*>& kernel_registries,
+                                          std::vector<std::unique_ptr<ComputeCapability>>& result) const {
+  // switch between mkldnn-vanilla and mkldnn-subgraph implementation using
+  // MKLDNN_SUBGRAPH environment variable
+  bool use_subgraph = true;
+
+  const char* env = getenv("ORT_MKLDNN_SUBGRAPH");
+  if (env != nullptr) {
```

Comment:
Even if the env variable is not present, use_subgraph will be true. Is that the intention?

Reply:
yes, I think we want to enable by default if all onnx models in our test .zip file show positive gains. (which looks like it does)
we still have a way to disable via env variable if needed. after some time, we will remove the old mkldnn kernel based impl and get rid of the env variable.